### PR TITLE
[LLT-5367] Update golang.com/x/net due to BDSA-2024-0942

### PR DESCRIPTION
**Pull request by Jauler**
### Problem
Update golang.com/x/net due to BDSA-2024-0942

### :ballot_box_with_check: Definition of Done checklist
- [x] Commit history is clean ([requirements](../blob/main/docs/git_commit_messages_requirements.md))
- [x] README.md is updated
- [x] Functionality is covered by unit or integration tests

### CODE REVIEW

**Review by tomaszklak — APPROVED** [review]
+2

File: wireguard-go-rust-wrapper/wireguard-go/go.sum [hunk 1]
```diff
@@ -5,10 +5,16 @@ github.com/google/uuid v1.5.0 h1:1p67kYwdtXjb0gL0BPiP1Av9wiZPo5A8z2cWkTZ+eyU=
 github.com/google/uuid v1.5.0/go.mod h1:TIyPZe4MgqvfeYDBFedMoGGpEw/LqOeaOT+nhxU+yHo=
 golang.org/x/crypto v0.18.0 h1:PGVlW0xEltQnzFZ55hkuX5+KLyrMYhHld1YHO4AKcdc=
 golang.org/x/crypto v0.18.0/go.mod h1:R0j02AL6hcrfOiy9T4ZYp/rcWeMxM3L6QYxlOuEG1mg=
```
Comment (stalowyjez) [inline wireguard-go-rust-wrapper/wireguard-go/go.sum:7]: Shouldn't these old versions be removed?

**Review by stalowyjez — APPROVED** [review]
+1

**Review by stalowyjez — APPROVED** [review]
lgtm also after rebase, +2

**Pull request merged**